### PR TITLE
Adding support for river via river-cpp/river-py packages

### PR DESCRIPTION
**Pull request by pbotros**
Adding pbotros/river packages, river-cpp and river-py packages for C++ library and Python bindings, respectively.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by pbotros** [comment]
@conda-forge-admin, please ping help-c-cpp

File: recipes/river-cpp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pbotros/river/archive/389b9ca6f76fdd2528aad96e67d33fc45a764d75.tar.gz
+  sha256: 7e4e6b19c7b284df1c6c8b500a342ad7d310a43d30f8f4e9ca3d987a9864b480
+
+build:
+  number: 0
```
Comment (beckermr) [inline recipes/river-cpp/meta.yaml:13]: We should add a run export giving the Abi range.

**Review by beckermr — CHANGES_REQUESTED** [review]
We should use a run export.

File: recipes/river-py/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  host:
+    - river-cpp {{ version }}
+    - python
+    - glog
+    - cython
+    - setuptools
+    - pip
+  run:
+    - python
+    - river-cpp
```
Comment (beckermr) [inline recipes/river-py/meta.yaml:31]:
```suggestion
```

Not needed now due to run export.

**Review by beckermr — APPROVED** [review]

**Comment by pbotros** [comment]
Thanks @beckermr for the prompt review and feedback!

**Pull request merged**